### PR TITLE
[Fix][multi-catalog]Fix coredump when reading the parquet file for multi-thread

### LINKED ISSUE #10634: [Bug] Coredump when reading the parquet file for multi-thread
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master branch

### What's Wrong?

when I run the tpch table sql for external hive table:`select count(*) from tpch1.lineitem`, and get this error.

```
*** Aborted at 1657008864 (unix time) try "date -d @1657008864" if you are using GNU date ***
*** SIGSEGV unkown detail explain (@0x0) received by PID 891067 (TID 0x7eff5c08e700) from PID 0; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /home/huangzhaowei.416/incubator-doris/be/src/common/signal_handler.h:407
 1# 0x00007F00107FA840 in /lib/x86_64-linux-gnu/libc.so.6
 2# parquet::arrow::FileReader::GetRecordBatchReader(std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::shared_ptr<arrow::RecordBatchReader>*) in /home/huangzhaowei.416/incubator-doris/output/be/lib/doris_be
 3# doris::ParquetReaderWrap::prefetch_batch() at /home/huangzhaowei.416/incubator-doris/be/src/exec/arrow/parquet_reader.cpp:538
 4# execute_native_thread_routine in /home/huangzhaowei.416/incubator-doris/output/be/lib/doris_be
 5# start_thread at /build/glibc-vjB4T1/glibc-2.28/nptl/pthread_create.c:487
 6# clone at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
```

### What You Expected?

Get the result.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SaintBacchus**
# Proposed changes

Issue Number: close #10634

## Problem Summary:
There is two issue fixed in this pr:

**The first issuse** is the C++ code rule of `do not call virtual function in constructor or deconstructor`. 
The deconstructor function of `ArrowReaderWrap` call the virtual function named `close()`.
When deconstructing, it will nerver call `ParquetReaderWrap::close()` just call the `ArrowReaderWrap::close()`

**The second issuse** is parallism deconstructing for `ParquetReaderWrap` and `prefetch_batch`.
`prefetch_batch` use `thread.detach()` to seprate the control from `ParquetReaderWrap`, but it rely on some local vars from `ParquetReaderWrap` such as **`_closed ` /`_total_groups ` and `_reader`**

In this case, `ParquetReaderWrap` may call deconstructor before `prefetch_batch` and then get the core dump.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — DISMISSED** [review]

**Review by morningman — APPROVED** [review]
